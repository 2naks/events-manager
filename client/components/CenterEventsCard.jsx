import React, { Component } from 'react';
import AdminEvent from './AdminEvent.jsx';

class CenterEventsCard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="card">
        <h6 className="card-header"><b>EVENTS</b></h6>
        {
          (this.props.events.map((event) => <AdminEvent key={event.id} event={event} />))
        }
      </div>
    );
  }
}

export default CenterEventsCard;
