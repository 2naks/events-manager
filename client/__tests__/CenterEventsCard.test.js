import CenterEventsCard  from '../components/CenterEventsCard.jsx';

describe('CenterEventsCard Component', () => {
  const props = {
      events:[{
        id: '1',
        title: 'Seminar',
        description: "",
        location: "",
        time:"5:00",
        date:'2020-01-01',
        image: "",
        Center:[]
      }]
    };

  describe('CenterEventsCard  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<CenterEventsCard {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

});
