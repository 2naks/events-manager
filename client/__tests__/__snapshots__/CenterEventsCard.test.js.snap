// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterEventsCard Component CenterEventsCard  should render self it should render for dumb component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterEventsCard
    events={
      Array [
        Object {
          "Center": Array [],
          "date": 1515504777425,
          "description": "",
          "id": "1",
          "image": "",
          "location": "",
          "time": "5:00",
          "title": "Seminar",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h6
          className="card-header"
        >
          <b>
            EVENTS
          </b>
        </h6>,
        Array [
          <AdminEvent
            event={
              Object {
                "Center": Array [],
                "date": 1515504777425,
                "description": "",
                "id": "1",
                "image": "",
                "location": "",
                "time": "5:00",
                "title": "Seminar",
              }
            }
          />,
        ],
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <b>
            EVENTS
          </b>,
          "className": "card-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "EVENTS",
          },
          "ref": null,
          "rendered": "EVENTS",
          "type": "b",
        },
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "event": Object {
            "Center": Array [],
            "date": 1515504777425,
            "description": "",
            "id": "1",
            "image": "",
            "location": "",
            "time": "5:00",
            "title": "Seminar",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h6
            className="card-header"
          >
            <b>
              EVENTS
            </b>
          </h6>,
          Array [
            <AdminEvent
              event={
                Object {
                  "Center": Array [],
                  "date": 1515504777425,
                  "description": "",
                  "id": "1",
                  "image": "",
                  "location": "",
                  "time": "5:00",
                  "title": "Seminar",
                }
              }
            />,
          ],
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              EVENTS
            </b>,
            "className": "card-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EVENTS",
            },
            "ref": null,
            "rendered": "EVENTS",
            "type": "b",
          },
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "event": Object {
              "Center": Array [],
              "date": 1515504777425,
              "description": "",
              "id": "1",
              "image": "",
              "location": "",
              "time": "5:00",
              "title": "Seminar",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
